server/src/game/client_actor.rs

```diff
@@ -0,0 +1,125 @@
+use std::time::{Duration, Instant};
+
+use actix::prelude::*;
+use actix_web_actors::ws;
+
+use super::messages::{Connect, Disconnect, ServerMessage};
+use super::server_actor::{SessionInfo, WsServer};
+
+const HEARTBEAT_INTERVAL: Duration = Duration::from_secs(5);
+const CLIENT_TIMEOUT: Duration = Duration::from_secs(10);
+
+pub struct WsClient {
+    game_name: Option<String>,
+    game_server: Addr<WsServer>,
+    last_heartbeat: Instant,
+    session_info: SessionInfo,
+}
+
+impl WsClient {
+    /// Creates a new [`WsClient`], an actor which sits between the user and the game server.
```

Comment:
The purpose of `WsClient` should be explained in the struct-level documentation, not the documentation for a specific function. Then, the documentation here can just be simplified to "Creates a new `WsClient`".